### PR TITLE
feat: RUM SDK auto injection

### PR DESCRIPTION
**Pull request by dmehala**
# Description
Basically [dmehala/rum-injection](https://github.com/DataDog/httpd-datadog/tree/dmehala/rum-injection) but rebased on `HEAD`. 

My goal is to merge what I currently working for RUM and endless rebase.

Left to do: 
- [x] Make the feature is optional.
- [x] Add directives configuration instead of JSON configuration.
- [x] Update private documentation.
- [x] Update configuration documentation.

### CODE REVIEW

File: doc/configuration.md [hunk 1]
```diff
@@ -151,3 +155,96 @@ When calling the `</foo>` endpoint, both the team and location tags will be adde
...
+
+## `DatadogRum` directive
+   - **Description**: Enable or disable the RUM Injection
+   - **Syntax:** DatadogRum *On\|Off*
+   - **Default:** On
+   - **Context:** Server and Directory
+
+Directive to enable or disable RUM SDK Injection.
+
+## `DatadogRumIncludeConfiguration` directive
```
Comment (dmehala) [inline doc/configuration.md:174]: Not true anymore.
- [ ] Update with new directive.

File: doc/configuration.md [hunk 2]
```diff
@@ -151,3 +155,95 @@ When calling the `</foo>` endpoint, both the team and location tags will be adde
...
+
+> [!IMPORTANT]
+> 🚧 RUM SDK Injection is currently in Private Beta! 🚧
+>
+> Interested in early access? [Apply to join the beta program today!](https://www.datadoghq.com/private-beta/rum-sdk-auto-injection/)
+
+## `DatadogRum` directive
+   - **Description**: Enable or disable the RUM Injection
+   - **Syntax:** DatadogRum *On\|Off*
+   - **Default:** On
```
Comment (pablomartinezbernardo) [inline doc/configuration.md:169]:
```suggestion
   - **Default:** Off
```

The code actually behaves like this, right?

**Review by pablomartinezbernardo — APPROVED** [review]
One small comment

File: doc/configuration.md (hunk 2 above)
Comment (dmehala) [reply doc/configuration.md:169]: Indeed

**Pull request merged**